Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverPlan.hs

```diff
@@ -113,9 +225,23 @@ updateCoinFieldsByDriverId driverId amount = do
 
 instance FromTType' BeamDF.DriverPlan DriverPlan where
   fromTType' BeamDF.DriverPlanT {..} = do
-    pure $
-      Just
-        DriverPlan
+    subscriptionServiceRelatedData <- do
+      case commodityData of
+        Just cd -> do
+          let parsedData = A.fromJSON cd :: A.Result SubscriptionServiceRelatedData
```

Comment:
Lets do this way..
Beam side add one column, `rentalVehicleNumber` which is Maybe.
But at Domain side, keep `SubscriptionServiceRelatedData` data type. 
In fromTType we can have a logic like this
```
case serviceName of 
 YATRI_RENTAL -> Vehicle rentalVehicleNumber
 YATRI_SUBSCRIPTION -> NoData
 ```
This will help both in analytics and Domain will also be clear.

Reply:
You can remove commodityData type also.